### ISSUE TITLE
something wrong happened in training part

### ISSUE THREAD

**Issue by hanguangmic** [hanguangmic/udacity_image_classifer#1]
# TODO: Build and train your network
def nn_model_setup(structure = 'vgg16', dropout = 0.2, hiden_layer1 = 120, lr = 0.001):
    if structure == 'vgg16':
        model = models.vgg16(pretrained = True)
    elif structure == 'densenet121':
        model = models.densenet121(pretrained = True)
    elif structure == 'alexnet':
        model = models.alexnet(pretrained = True)
    else:
        print ('Please select from vgg16, densenet121 or alexnet')    
    for para in model.parameters():
        para.requires_grad = False
        from collections import OrderedDict
        model.classifier = nn.Sequential(OrderedDict([
            ('dropout',nn.Dropout(dropout)),
            ('inputs', nn.Linear(model_selection[structure], hiden_layer1)),
            ('relu1', nn.ReLU()),
            ('hidden_layer1', nn.Linear(hiden_layer1, 90)),
            ('relu2',nn.ReLU()),
            ('hidden_layer2',nn.Linear(90,80)),
            ('relu3',nn.ReLU()),
            ('hidden_layer3',nn.Linear(80,102)),
            ('output', nn.LogSoftmax(dim=1))
                          ]))
    
    criterion = nn.NLLLoss()
    optimizer = optim.Adam(model.classifier.parameters(), lr)
    model.to(device)
    return model, criterion, optimizer

model, criterion, optimizer = nn_model_setup('densenet121')

epoches = 10
training_loss =0 
running_loss = 0 
steps = 0
print_every = 5
for e in range (epoches):
    for i, (images, labels) in enumerate(training_dataloaders):
        steps += 1
        images, labels = images.to(device), labels.to(device)
        optimizer.zero_grad()
        log_predictions = model.forward(images)
        loss = criterion(log_predictions, labels)
        loss.backward()
        optimizer.step()
        running_loss += loss.item()
    if steps % print_every == 0:    
        model.eval()
        accuracy = 0 
        testing_loss = 0
        
        for ii, (inputs2,labels2) in enumerate(testing_dataloaders):
            optimizer.zero_grad()
            test_images, test_labels = inputs2.to(device), labels2.to(device)
            model.to('cuda:0')
            with torch.no_grad():
                log_test_predictions = model.forward(test_images)
                test_predictions = torch.exp(log_test_predictions)
                testing_loss += criterion(log_test_predictions, test_labels).item()
                topk, topc = test_predictions.topk(1, dim =1)
                equals = topc == test_labels.view(*topc.shape)
                accuracy += torch.mean(equals.type(torch.FloatTensor)).item()
            
        print (f"Epoch {e+1}/{epoches}.. "
                  f"Train loss: {running_loss/print_every:.3f}.. "
                  f"Test loss: {testing_loss/len(testing_dataloaders):.3f}.. "
                  f"Test accuracy: {accuracy/len(testing_dataloaders):.3f}")
        model.train()
        running_loss = 0
